js/modules/k6/ws/ws.go

```diff
@@ -360,9 +360,15 @@ func (s *Socket) Send(message string) {
 }
 
 // SendBinary writes the given ArrayBuffer message to the connection.
-func (s *Socket) SendBinary(message *goja.ArrayBuffer) {
-	if err := s.conn.WriteMessage(websocket.BinaryMessage, message.Bytes()); err != nil {
-		s.handleEvent("error", common.GetRuntime(s.ctx).ToValue(err))
+func (s *Socket) SendBinary(message goja.Value) {
+	msg := message.Export()
+	if ab, ok := msg.(goja.ArrayBuffer); ok {
+		if err := s.conn.WriteMessage(websocket.BinaryMessage, ab.Bytes()); err != nil {
+			s.handleEvent("error", common.GetRuntime(s.ctx).ToValue(err))
+		}
+	} else {
+		common.Throw(common.GetRuntime(s.ctx),
+			fmt.Errorf("expected ArrayBuffer as argument, received: %T", msg))
```

Comment:
Ideally, this should be the type of `message` but that will require a lot more code and practically can have a case where we can get one (if it's nil/undefined :thinking: )

I did try it and 
```golang
message.ToObject(common.GetRuntime(s.ctx)).ClassName()
```
seems to work okay and blows up with TypeError if `message` is null/undefined which is more or less what we want so ... maybe we can just add it ? But I would prefer if more testing is done that it won't panic the whole VU in some strange case

Reply:
I'm not sure if we should bother with that. Sure, it would be good to log the JS type instead, but not everything can be converted to an object. For example, `socket.sendBinary(1)` throws `expected ArrayBuffer as argument, received: string` for some reason, which is more confusing than `expected ArrayBuffer as argument, received: int64`. And we'd have to handle the `null`/`undefined` case as well.

I don't see any way in the Go API to call `typeof` so we'd have to `RunString()` it. Do you think all this is worth it?

Reply:
OK, should be resolved by 339af11c. I think I was exhaustive in the error tests, but let me know if I missed anything.

Also I rebased on `master` and cleaned up the fixup commit, but didn't want to squash the changes for your suggestions yet, in case @na-- disagrees with it.